templates/common/gcp/files/etc-kubernetes-credential-providers-gcr-credential-provider.yaml

```diff
@@ -0,0 +1,14 @@
+mode: 0644
+path: "/etc/kubernetes/credential-providers/gcr-credential-provider.yaml"
+contents:
+  inline: |
+    apiVersion: kubelet.config.k8s.io/v1
+    kind: CredentialProviderConfig
+    providers:
+      - name: gcr-credential-provider
+        matchImages:
+          - "gcr.io"
+          - "*.gcr.io"
+          - "container.cloud.google.com"
+        defaultCacheDuration: "0s"
```

Comment:
Why 0s on this one? Are the others not cached for 5s?

Reply:
I read the code for the gcp auth provider, and they default to 0s cache duration (unless it's overridden by an env var): 

https://github.com/openshift/cloud-provider-gcp/blob/master/cmd/auth-provider-gcp/provider/provider.go#L76-L84 

This apparently mirrors the in tree provider, so i figured leaving it set to 0s was fine.